draft-ietf-protected-audience-key-value-server-api.md

```diff
@@ -1,78 +1,175 @@
+Responses.
 
-*   Each request contains one or more partitions. Each partition is a collection of keys that can be processed together by the service without any potential privacy leakage. Keys from one interest group must be in the same partition. Keys from different interest groups with the same joining site may or may not be in the same partition, so the server User Defined Functions should not make any assumptions based on that.
-*   Each partition contains one or more key groups. Each key group has its unique attributes among all key groups in the partition. The attributes are represented by a list of “Tags”. Besides tags, the key group contains a list of keys to look up.
-*   Each partition has a unique id.
-*   Each partition has a compression group field. Results of partitions belonging to the same compression group can be compressed together in the response. Different compression groups must be compressed separately. See more details below. The expected use case by the client is that interest groups from the same joining origin and owner can be in the same compression group.
+# Message Format Specifications {#format}
 
-TODO
+## Overview
+
+On a high level, the Key Value Service adheres to the following communication protocol:
+
+-   Data is transmitted over HTTPS using the `POST` method.
+-   Data within the request and response is encrypted with [HPKE].
+-   The core request and response data is in [CBOR].
+
+### Encryption {#encryption}
+
+The Key Value Service uses [HPKE] with the following configuration for encryption:
+
+-   KEM (Key encapsulation mechanisms): 0x0020 DHKEM(X25519, HKDF-SHA256), see
+    {{Section 1 of HPKE}}
+-   KDF (key derivation functions): 0x0001 HKDF-SHA256, see {{Section 7.2 of HPKE}}
+-   AEAD (Authenticated Encryption with Associated Data): AES-256-GCM, see
+    {{Section 7.3 of HPKE}}
+
+The server is repurposing the [OHTTP] encapsulation mechanism ({{Section 4.6 of OHTTP}}), new
+media types need to be defined:
+
+-   The OHTTP request media type is `message/ad-auction-trusted-signals-request`
+-   The OHTTP response media type is `message/ad-auction-trusted-signals-response`
+
+These media types are concatenated with other fields when creating the [HPKE] encryption context.
+
+### Message Framing {#framing}
+
+Inside the ciphertext, the requests and responses have the following framing:
+
+| Byte     | 0         | 0             | 1 to 4   | 5 to Size+4       | Size+5 to end   |
+| -------- | --------- | ------------- | -------- | ----------------- | --------------- |
+| Bits     | 7-2       | 1-0           | \*       | \*                | \*              |
+| -------- | --------- | ------------- | -------- | ----------------- | --------------- |
+| Contents | Unused    | Compression   | Size     | Request Payload   | Padding         |
+
+The request/response is framed with a 5 byte header.
+
+The first byte is the format+compression byte. The lower 2 bits are used for compression
```

Comment:
This should link to the #compression section.

Reply:
Done.